proxy.go

```diff
@@ -69,22 +76,17 @@ func (ph ProxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 func (ph ProxyHandler) handleConnect(w http.ResponseWriter, req *http.Request) {
 	// Establish a connection to the server, or an upstream proxy.
-	u, err := ph.transport.Proxy(req)
 	id := req.Context().Value(contextKeyID)
-	if err != nil {
-		log.Printf("[%d] Error finding proxy for %v: %v", id, req.Host, err)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
 	var server net.Conn
-	if u == nil {
+	var err error
+	if proxy, _ := ph.transport.Proxy(req); proxy == nil {
 		server, err = net.Dial("tcp", req.Host)
 	} else {
-		server, err = connectViaProxy(req, u.Host, ph.auth)
+		server, err = connectViaProxy(req, proxy.Host, ph.auth)
 		var dialErr *dialError
 		if errors.As(err, &dialErr) {
-			log.Printf("[%d] Temporarily blocking unreachable proxy: %q", id, u.Host)
-			ph.block(u.Host)
+			log.Printf("[%d] Temporarily blocking proxy: %q", id, proxy.Host)
+			ph.block(proxy.Host)
 		}
 	}
 	if err != nil {
```

Comment:
On the subject of logging errors, this error does not get logged.

Reply:
Ah you're right - if it goes down the `if` branch, then it won't get logged. But if it goes down the `else` branch, then it'll get logged inside `connectViaProxy()`. To try to keep things a bit consistent, I've added a new function called `connectDirect()` which logs the error, if one occurs.